forrest.go

```diff
@@ -27,31 +84,226 @@ itter indicates weather to use iterative splitting for all categorical features
+	}
+
+	for i, tree := range f.Trees {
+		tree.Root.Recurse(func(n *Node, cases []int, depth int) {
+			if n.Left == nil && n.Right == nil {
+				for _, j := range cases {
+					if val, err := strconv.ParseFloat(n.Pred, 64); err == nil {
+						predictions[j][i] = val
+					} else {
+						predictions[j][i] = math.NaN()
+					}
+				}
+			}
+		}, fm, cases, 0)
+	}
+
+	return predictions
+}
+
+func (f *Forest) Predict(fm *FeatureMatrix) []float64 {
+	n := fm.Data[0].Length()
+	bb := NewNumBallotBox(n)
+	for _, tree := range f.Trees {
+		tree.Vote(fm, bb)
+	}
+
+	preds := make([]float64, n)
+	for i := 0; i < n; i++ {
+		pred, _ := strconv.ParseFloat(bb.Tally(i), 64)
+		preds[i] = pred
+	}
+	return preds
+}
+
+// Prediction consists of a predicted Value and it's associated variance
+type Prediction struct {
+	Value    float64
+	Variance float64
+}
+
+// JackKnife estimates the variance of the predicted value from the RandomForest
+// using an infinitesimal jackknife estimator for the variance.
+//
+// predictionSlice is a list of the complete predictions for each observation
+// inbag is the inbag samples used by the RandomForest
+//
+// returns a Prediction for each value in the predictionSlice
+func JackKnife(predictionSlice, inbag [][]float64) ([]*Prediction, error) {
+	if len(predictionSlice) == 0 || len(inbag) == 0 {
+		return nil, fmt.Errorf("prediction and inbag size must be equal")
+	}
+
+	var (
+		m    = len(predictionSlice)
+		n    = len(inbag)
+		B    = float64(len(predictionSlice[0]))
+		B2   = B * B
+		nvar = avgVar(inbag)
+	)
+
+	var (
+		avgPreds = make([]float64, m)
+		sums     = make([]float64, m)
+		output   = make([]*Prediction, m)
+		errs     = make(chan error, m)
+		wg       sync.WaitGroup
+	)
+
+	// normalize the prediction slices
+	for i, predictions := range predictionSlice {
+		wg.Add(1)
+		go func(idx int, p []float64) {
+			defer wg.Done()
+
+			preds := govector.Vector(p)
+			avgPred := preds.Mean()
+
+			predictionSlice[idx] = preds.Apply(func(f float64) float64 { return f - avgPred })
+			avgPreds[idx] = avgPred
+		}(i, predictions)
+	}
+	wg.Wait()
+
+	// calculate the raw infinitesimal jackknife values
+	// V_{i, j} = \sum_{i, n} Cov( inbag[i], pred ) ^ 2
+	for i, preds := range predictionSlice {
+		wg.Add(1)
+		go func(idx int, p []float64) {
+			defer wg.Done()
+			sum := 0.0
+			for i := 0; i < n; i++ {
+				val, err := govector.DotProduct(govector.Vector(inbag[i]), p)
+				if err != nil {
+					errs <- err
+					return
+				}
+
+				val *= val
+				val /= B2
+				sum += val
+			}
+			sums[idx] = sum
+		}(i, preds)
+	}
+	wg.Wait()
+
+	// check for errors
+	if len(errs) > 0 {
+		return nil, <-errs
+	}
+
+	// normalize the IJ value with Monte-Carlo bias correction
+	for i, preds := range predictionSlice {
+		wg.Add(1)
+		go func(idx int, p []float64) {
+			defer wg.Done()
+			bv := 0.0
+			for i := 0; i < len(p); i++ {
+				bv += p[i] * p[i]
+			}
+
+			variance := sums[idx] - float64(n)*nvar*(bv/B)/B
```

Comment:
http://arxiv.org/pdf/1311.4555.pdf . I'll add a comment in the code too. 

`R` code is here also : https://github.com/swager/randomForestCI (look at `infJack`)